gointelowl/job.go

```diff
@@ -0,0 +1,242 @@
+package gointelowl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type UserDetails struct {
+	Username string `json:"username"`
+}
+
+type Report struct {
+	Name                 string                 `json:"name"`
+	Status               string                 `json:"status"`
+	Report               map[string]interface{} `json:"report"`
+	Errors               []string               `json:"errors"`
+	ProcessTime          float64                `json:"process_time"`
+	StartTime            time.Time              `json:"start_time"`
+	EndTime              time.Time              `json:"end_time"`
+	RuntimeConfiguration map[string]interface{} `json:"runtime_configuration"`
+	Type                 string                 `json:"type"`
+}
+
+type Job struct {
+	ID                       int                    `json:"id"`
+	User                     UserDetails            `json:"user"`
+	Tags                     []Tag                  `json:"tags"`
+	ProcessTime              float64                `json:"process_time"`
+	AnalyzerReports          []Report               `json:"analyzer_reports,omitempty"`
+	ConnectorReports         []Report               `json:"connector_reports,omitempty"`
+	Permission               map[string]interface{} `json:"permission,omitempty"`
+	IsSample                 bool                   `json:"is_sample"`
+	Md5                      string                 `json:"md5"`
+	ObservableName           string                 `json:"observable_name"`
+	ObservableClassification string                 `json:"observable_classification"`
+	FileName                 string                 `json:"file_name"`
+	FileMimetype             string                 `json:"file_mimetype"`
+	Status                   string                 `json:"status"`
+	AnalyzersRequested       []string               `json:"analyzers_requested" `
+	ConnectorsRequested      []string               `json:"connectors_requested"`
+	AnalyzersToExecute       []string               `json:"analyzers_to_execute"`
+	ConnectorsToExecute      []string               `json:"connectors_to_execute"`
+	ReceivedRequestTime      *time.Time             `json:"received_request_time"`
+	FinishedAnalysisTime     *time.Time             `json:"finished_analysis_time"`
+	Tlp                      string                 `json:"tlp"`
```

Comment:
Can this be made an enum? Let's create an issue for this.

Reply:
True! I'll create an issue of this.